fontforge/woff.c

```diff
@@ -348,11 +348,43 @@ return( NULL );
     }
 
     if ( sf!=NULL && metaOffset!=0 ) {
+	/*
+	* Boundary/integer overflow checks:
+	*
+	* We don't want to actually dereference a null pointer (returned
+	* by asking to allocate too much RAM) and we don't want to allocate
+	* a 0-sized chunk (caused when one of the (metaLenxxx + 1) values overflows).
+	*
+	* We can safely pass sf->woffMetadata as a NULL pointer because
+	* it's never accessed anywhere else without a check for it being
+	* NULL first
+	*/
+	if(metaLenUncompressed == 0xffffffff) {
```

Comment:
This is definitely an issue, though. 0xFFFFFFFF means two different things signed and unsigned. It might be worth redefining `metaLenUncompressed` and `metLenCompressed` to `uint32_t`s and casting the result from `getlong`.